managedplugin/version_checker.go

```diff
@@ -0,0 +1,84 @@
+package managedplugin
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Masterminds/semver"
+	cloudquery_api "github.com/cloudquery/cloudquery-api-go"
+	"github.com/rs/zerolog"
+)
+
+type PluginVersionWarner struct {
+	hubClient *cloudquery_api.ClientWithResponses
+	logger    zerolog.Logger
+}
+
+func NewPluginVersionWarner(logger zerolog.Logger) (*PluginVersionWarner, error) {
+	hubClient, err := getHubClient(logger, HubDownloadOptions{}) // Does not use auth token, since API call is public
```

Comment:
Not a blocker, but for tracking and rate limiting purposes on the backend we might want to use an auth token if available (e.g. the user is logged in).
Unauthenticated requests can have lower rate limits

Reply:
Added optional support for AuthToken.

Reply:
Why not get `hubClient *cloudquery_api.ClientWithResponses` as a param?